lib/endpoints/class-wp-rest-controller.php

```diff
@@ -3,111 +3,113 @@
 
 abstract class WP_REST_Controller {
 
+	protected $route_base = '';
+	protected $id_regex    = '[\d]+';
+	protected $namespace   = '';
 	/**
 	 * Register the routes for the objects of the controller.
 	 */
 	public function register_routes() {
-		_doing_it_wrong( 'WP_REST_Controller::register_routes', __( 'The register_routes() method must be overriden' ), 'WPAPI-2.0' );
-	}
 
-	/**
-	 * Get a collection of items
-	 *
-	 * @param WP_REST_Request $request Full data about the request.
-	 * @return WP_Error|WP_REST_Response
-	 */
-	public function get_items( $request ) {
-		return new WP_Error( 'invalid-method', __( 'Method not implemented. Must be over-ridden in subclass.' ), array( 'status' => 405 ) );
-	}
+		$multiple_args = array();
+		$schema = $this->get_item_schema();
 
-	/**
-	 * Get one item from the collection
-	 *
-	 * @param WP_REST_Request $request Full data about the request.
-	 * @return WP_Error|WP_REST_Response
-	 */
-	public function get_item( $request ) {
-		return new WP_Error( 'invalid-method', __( 'Method not implemented. Must be over-ridden in subclass.' ), array( 'status' => 405 ) );
-	}
+		if ( method_exists( $this, 'get_items' ) ) {
 
-	/**
-	 * Create one item from the collection
-	 *
-	 * @param WP_REST_Request $request Full data about the request.
-	 * @return WP_Error|WP_REST_Request
-	 */
-	public function create_item( $request ) {
-		return new WP_Error( 'invalid-method', __( 'Method not implemented. Must be over-ridden in subclass.' ), array( 'status' => 405 ) );
-	}
+			$args = array(
+				'methods'         => WP_REST_Server::READABLE,
+				'callback'        => array( $this, 'get_items' ),
+			);
 
-	/**
-	 * Update one item from the collection
-	 *
-	 * @param WP_REST_Request $request Full data about the request.
-	 * @return WP_Error|WP_REST_Request
-	 */
-	public function update_item( $request ) {
-		return new WP_Error( 'invalid-method', __( 'Method not implemented. Must be over-ridden in subclass.' ), array( 'status' => 405 ) );
-	}
+			if ( method_exists( $this, 'get_items_permissions_check' ) ) {
```

Comment:
How do we handle the magical — if this is misspelled, we expose a bunch of data?

We could force override by having the default registry return false? e.g. you always have to successfully override.